File: WatchMyCalories/WatchMyCalories/LogFoodSheet.swift
import SwiftUI

struct LogFoodSheet: View {
    var onScanFood: () -> Void
    var onChooseFromLibrary: () -> Void
    var onLogManually: () -> Void

    var body: some View {
        VStack(spacing: 0) {
            // Handle bar
            Capsule()
                .fill(Color.secondary.opacity(0.3))
                .frame(width: 36, height: 5)
                .padding(.top, 8)
                .padding(.bottom, 20)

            Text("Log Food")
                .font(.system(.title3, design: .serif, weight: .bold))
                .foregroundStyle(Color.cwPrimary)
                .padding(.bottom, 20)

            VStack(spacing: 12) {
                logOptionButton(
                    title: "Scan Food",
                    subtitle: "Take a photo of your meal",
                    icon: "camera.fill",
                    action: onScanFood
                )

                logOptionButton(
                    title: "Choose from Library",
                    subtitle: "Select a photo from your library",
                    icon: "photo.on.rectangle",
                    action: onChooseFromLibrary
                )

                logOptionButton(
                    title: "Log Manually",
                    subtitle: "Enter food details by hand",
                    icon: "square.and.pencil",
                    action: onLogManually
                )
            }
            .padding(.horizontal)

            Spacer()
        }
        .presentationDetents([.medium])
    }

    private func logOptionButton(title: String, subtitle: String, icon: String, action: @escaping () -> Void) -> some View {
        Button(action: action) {
            HStack(spacing: 16) {
                Image(systemName: icon)
                    .font(.title2)
                    .foregroundStyle(Color.cwPrimary)
                    .frame(width: 44, height: 44)
                    .background(Color.cwPrimary.opacity(0.1))
                    .clipShape(RoundedRectangle(cornerRadius: 12, style: .continuous))

                VStack(alignment: .leading, spacing: 2) {
                    Text(title)
                        .font(.body)
                        .fontWeight(.semibold)
                        .foregroundStyle(Color.cwTextPrimary)
                    Text(subtitle)
                        .font(.caption)
                        .foregroundStyle(.secondary)
                }

                Spacer()

                Image(systemName: "chevron.right")
                    .font(.caption)
                    .foregroundStyle(.tertiary)
            }
            .padding()
            .background(Color.cwSurface)
            .clipShape(RoundedRectangle(cornerRadius: 16, style: .continuous))
            .shadow(color: Color.black.opacity(0.04), radius: 4, x: 0, y: 2)
        }
        .buttonStyle(.plain)
    }
}

File: WatchMyCaloriesAndroid/app/src/main/java/com/pning80/watchmycalories/ui/logfood/LogFoodSheet.kt
package com.pning80.watchmycalories.ui.logfood

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.KeyboardArrowRight
import androidx.compose.material.icons.filled.Edit
import androidx.compose.material.icons.filled.PhotoCamera
import androidx.compose.material.icons.filled.PhotoLibrary
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.unit.dp
import com.pning80.watchmycalories.ads.BannerAdView
import com.pning80.watchmycalories.ui.theme.Spacing

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun LogFoodSheet(
    onScanFood: () -> Unit,
    onChooseFromLibrary: () -> Unit,
    onLogManually: () -> Unit,
    onDismiss: () -> Unit
) {
    ModalBottomSheet(
        onDismissRequest = onDismiss,
        shape = RoundedCornerShape(topStart = Spacing.cardCorner, topEnd = Spacing.cardCorner),
        scrimColor = Color.Black.copy(alpha = 0.6f)
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = Spacing.pageHorizontal)
                .padding(bottom = 32.dp),
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.spacedBy(Spacing.cardGap)
        ) {
            Text(
                text = "Log Food",
                style = MaterialTheme.typography.titleLarge.copy(fontFamily = FontFamily.Serif),
                fontWeight = androidx.compose.ui.text.font.FontWeight.Bold,
                color = MaterialTheme.colorScheme.primary,
                modifier = Modifier.padding(bottom = Spacing.s)
            )

            // Icons mirror iOS LogFoodSheet.swift: camera.fill / photo.on.rectangle
            // / square.and.pencil → PhotoCamera / PhotoLibrary / Edit.
            LogOptionButton(
                title = "Scan Food",
                subtitle = "Take a photo of your meal",
                icon = Icons.Filled.PhotoCamera,
                onClick = onScanFood,
                testTag = "logFood_scanFood"
            )

            LogOptionButton(
                title = "Choose from Library",
                subtitle = "Select a photo from your library",
                icon = Icons.Filled.PhotoLibrary,
                onClick = onChooseFromLibrary,
                testTag = "logFood_chooseLibrary"
            )

            LogOptionButton(
                title = "Log Manually",
                subtitle = "Enter food details by hand",
                icon = Icons.Filled.Edit,
                onClick = onLogManually,
                testTag = "logFood_logManually"
            )

            // Banner ad — mirrors iOS LogFoodSheet.swift:56 (BannerAdView at the
            // bottom of the sheet). insetHorizontal=false: the sheet Column
            // already applies pageHorizontal padding.
            BannerAdView(insetHorizontal = false)
        }
    }
}

@Composable
private fun LogOptionButton(
    title: String,
    subtitle: String,
    icon: ImageVector,
    onClick: () -> Unit,
    testTag: String
) {
    Card(
        onClick = onClick,
        modifier = Modifier
            .fillMaxWidth()
            .testTag(testTag),
        colors = CardDefaults.cardColors(containerColor = MaterialTheme.colorScheme.surface),
        shape = RoundedCornerShape(Spacing.l)
    ) {
        Row(
            modifier = Modifier.padding(Spacing.l),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(Spacing.l)
        ) {
            Surface(
                shape = RoundedCornerShape(Spacing.m),
                color = MaterialTheme.colorScheme.primary.copy(alpha = 0.1f),
                modifier = Modifier.size(44.dp)
            ) {
                Box(contentAlignment = Alignment.Center, modifier = Modifier.fillMaxSize()) {
                    Icon(
                        imageVector = icon,
                        contentDescription = title,
                        tint = MaterialTheme.colorScheme.primary
                    )
                }
            }

            Column(modifier = Modifier.weight(1f)) {
                Text(
                    text = title,
                    style = MaterialTheme.typography.bodyLarge,
                    // iOS LogFoodSheet.swift option title is .body .semibold — match it.
                    fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold,
                    color = MaterialTheme.colorScheme.onSurface
                )
                Text(
                    text = subtitle,
                    style = MaterialTheme.typography.bodySmall,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
            }

            Icon(
                Icons.AutoMirrored.Filled.KeyboardArrowRight,
                contentDescription = null,
                tint = MaterialTheme.colorScheme.onSurfaceVariant,
            )
        }
    }
}
